backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,75 @@
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import CollectionMetadata, CollectionVersion, DatasetArtifact, DatasetMetadata, DatasetStatus
+
+
+class DatabaseProviderInterface:
+        
+    def create_collection(self, owner: str, collection_metadata: CollectionMetadata):
+        pass
+
+    def get_collection(self, collection_id: str) -> CollectionVersion:
+        pass
+
+    def get_all_collections(self) -> Iterable[CollectionVersion]:  # TODO: add filters if needed
+        pass
+
+    def delete_collection(self, collection_id: str) -> None:
+        pass
+
+    def save_collection_metadata(self, version_id: str, collection_metadata: CollectionMetadata) -> None:
```

Comment:
this is to be used for saving repeated edits to a collection version while it remains unpublished, right?